.github/actions/prepare/action.yml

```diff
@@ -24,3 +24,8 @@ runs:
       run: |
         pnpm install-deps
       shell: bash
+
+    - name: Install e2e dependencies
+      run: |
+        cd e2e && pnpm install
```

Comment:
You probably need to use frozen lock file instead of updating the lock file, Similar to install-deps.